pkgs/development/python-modules/bootstrapped-pytest/default.nix

```diff
@@ -0,0 +1,147 @@
+# A pytest(-xdist) package that does not depend on any other python packages
+# Can be used to test all other packages without causing infinite recursion
+
+{ bootstrapped-pip
+, fetchPypi
+, isPy27
+, lib
+, makeWrapper
+, python
+, stdenv
+, unzip
+
+# python packages to take sources from
+, apipkg
+, attrs
+, execnet
+, iniconfig
+, packaging
+, pluggy
+, py
+, pyparsing
+, pytest
+, pytest-forked
+, pytest-xdist
+, setuptools-scm
+, setuptools
+, toml
+, python2Packages
+, breakpointHook
+}:
+
+let
+
+  # the order is relevant for renaming unpacked archives
+  allDeps = [
+    apipkg
+    attrs
+    execnet
+    iniconfig
+    packaging
+    pluggy
+    py
+    pyparsing
+    pytest
+    pytest-forked
+    pytest-xdist
+    setuptools-scm
+    setuptools
+    toml
+  ] ++ (lib.optionals isPy27 ( with python2Packages; [
+    atomicwrites
+    configparser
+    contextlib2
+    funcsigs
+    importlib-metadata
+    more-itertools
+    pathlib2
+    scandir
+    six
+    zipp
+  ]));
+
+  extraPnames = lib.optionals isPy27 [
+    "atomicwrites"
+    "configparser"
+    "contextlib2"
+    "funcsigs"
+    "importlib_metadata"
+    "more-itertools"
+    "pathlib2"
+    "scandir"
+    "six"
+    "zipp"
+  ];
+
+in
+
+stdenv.mkDerivation rec {
+
+  pname = "pytest";
+
+  inherit (pytest) version;
+
+  name = "${python.libPrefix}-bootstrapped-${pname}-${version}";
+
+  srcs = lib.forEach allDeps (dep: dep.src);
+
+  allDepNames = lib.forEach allDeps (dep: dep.pname);
+
+  sourceRoot = ".";
+
+  format = "other";
+
+  doCheck = true;
+
+  postPatch = ''
+    mkdir -p $out/bin
+  '';
+
+  nativeBuildInputs = [ bootstrapped-pip makeWrapper unzip  breakpointHook];
+
+  buildInputs = [ python ];
+
+  dontInstall = true;
+
+  buildPhase = lib.strings.optionalString (!stdenv.hostPlatform.isWindows) ''
```

Comment:
Maybe make this the internal function I mentioned above for creating a Python environment using pip? In time we can see if it can be upgraded to something for external use as well.